src/coordinator.coffee

```diff
@@ -244,6 +249,17 @@ class Coordinator extends EventEmitter
         delete @process[k]
       return callback null, processes
 
+  updateParticipantMetadata: (node, metadata, callback) ->
+    metadata = {} unless metadata
+    process = null
+    for k, v of @processes
+      if k == node
+        process = v
+    unless process
+      return callback new Error "Node #{node} not defined"
+    @processes[k].metadata = metadata
```

Comment:
This is maybe not quite right... `@processes` are the `child_process` instances, for (and only for) the things which the coordinator starts. Things which are discovered don't show up here, but may still have metadata attached? 
Also affects roles which are ignored...

But maybe we cannot do much about this until we have a clearer split of actual/desired states.